spa/src/components/common/Modal/MaxPagesPublishedModal/MaxPagesPublishedModal.tsx

```diff
@@ -0,0 +1,65 @@
+import PropTypes, { InferProps } from 'prop-types';
+import { Button, Link, Modal, ModalFooter, ModalHeader } from 'components/base';
+import { PRICING_URL } from 'constants/helperUrls';
+import { PLAN_NAMES } from 'constants/orgPlanConstants';
+import { EnginePlan } from 'hooks/useContributionPage';
+import { ModalContent, ModalHeaderIcon, RedEmphasis } from './MaxPagesPublishedModal.styled';
+
+const MaxPagesPublishedPropTypes = {
+  currentPlan: PropTypes.oneOf(Object.keys(PLAN_NAMES).filter((name) => name !== 'PLUS')).isRequired,
+  onClose: PropTypes.func.isRequired,
+  open: PropTypes.bool
+};
+
+export interface MaxPagesPublishedModalProps
+  extends Omit<InferProps<typeof MaxPagesPublishedPropTypes>, 'currentPlan'> {
+  currentPlan: Omit<EnginePlan['name'], 'PLUS'>;
+}
+
+export function MaxPagesPublishedModal({ currentPlan, onClose, open }: MaxPagesPublishedModalProps) {
+  return (
+    <Modal open={!!open}>
+      <ModalHeader icon={<ModalHeaderIcon />} onClose={onClose}>
+        <RedEmphasis>Max Pages Published</RedEmphasis>
+      </ModalHeader>
+      <ModalContent>
+        {currentPlan === 'FREE' && (
```

Comment:
I think it would be a good idea to create a constant file for the plans, we are using it in a few places now.
```suggestion
        {currentPlan === ORG_PLAN.FREE && (
```

```js
const ORG_PLAN = {
  FREE: 'FREE',
  CORE: 'CORE',
  PLUS: 'PLUS',
}
```

Reply:
I like this a lot, updated in a01be2677b0e736290f1525d14e5f457e0d12d90. I generated the names dynamically so that PLAN_NAMES and PLAN_LABELS can't fall out of sync.